src/assert.alp

```diff
@@ -2,9 +2,29 @@
  -}
 module assert
 
-export equal
+export equal/2, isTrue/2, isFalse/2
 
+-- It would be nice to convert this into a pure type rather
+-- than a dummy type with throwing used to signify failure.
```

Comment:
I'm not sure I entirely follow since we need exceptions for eunit but I may be getting hung up on implementation/integration needs  (I've also only had one coffee so far today)

Are you looking for something that unifies the failure effect and success?  Or something that wraps a pure Alpaca pass/fail type that then gets converted to exceptions in code generation just for eunit?  (presumably so that it could get handled separately by some other hypothetical test runner)

Thanks!

Reply:
That sort of thing! A test being a zero arity function that returns some data structure representing success and failure and then the `test` construct throws or does not depending on what is returned. To me it feels like it would be more fitting for an ML language to work with values and pure functions rather than exceptions.

Would also give us a clear way to make multiple assertions in a single test.

```ocaml
module assert

val all : list (fn () -> assertion) -> assertion
let all assertions =
  assertions
    |> list.find (fn testCase -> isFail (testCase ()))
    |> option.orElse Pass
```

Reply:
Interesting, I do like the idea of chained assertions like this.  Maybe this is a larger testing conversation we could open up in the mailing list or an Alpaca issue?  I kind of want simple fixture-with-sequence-of-test-cases things too.